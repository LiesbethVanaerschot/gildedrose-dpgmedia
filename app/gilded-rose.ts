export class Item {
  name: string;
  sellIn: number;
  quality: number;

  constructor(name, sellIn, quality) {
    this.name = name;
    this.sellIn = sellIn;
    this.quality = quality;
  }
}

export class GildedRose {
  items: Array<Item>;

  constructor(items = [] as Array<Item>) {
    this.items = items;
  }

  updateQuality() {
    for (let i = 0; i < this.items.length; i++) {
      //first exception Sulfuras is a stable product
      if (this.items[i].name !== 'Sulfuras, Hand of Ragnaros') {
        //normal case: per update sellIn -1
        this.items[i].sellIn = this.items[i].sellIn - 1;

        //If-else statement that separates the ones that decrease from the ones that increase in quality.
        if ( this.items[i].name === 'Aged Brie' || this.items[i].name === 'Backstage passes to a TAFKAL80ETC concert') {
          //standard + 1 in quality
          this.items[i].quality = this.items[i].quality < 50 ? this.items[i].quality + 1 : 50;
          //but backstage passes have two extra levels till the event has passed
          if (this.items[i].name === 'Backstage passes to a TAFKAL80ETC concert') {
            if (this.items[i].sellIn > 5 && this.items[i].sellIn < 11) {
              this.items[i].quality = this.items[i].quality + 1;
            } else if (this.items[i].sellIn < 6) {
              this.items[i].quality = this.items[i].quality + 2;
            } else if (this.items[i].sellIn < 0) {
              this.items[i].quality = 0;
            }
          }
        } else {
          //Conjured cake degrade twice as fast as the rest, Quality is never negative
          if (this.items[i].sellIn > 0) {
            this.items[i].quality = this.items[i].quality > 0 ? this.items[i].name === 'Conjured Mana Cake' ? this.items[i].quality - 2 : this.items[i].quality - 1 : 0;
          } else {
            this.items[i].quality = this.items[i].quality > 0 ? this.items[i].name === 'Conjured Mana Cake' ? this.items[i].quality - 4 : this.items[i].quality - 2 : 0;
          }
        }
      }
    }

      //nog even laten staan
      /* if (this.items[i].name != 'Aged Brie' && this.items[i].name != 'Backstage passes to a TAFKAL80ETC concert') {
        if (this.items[i].quality > 0) {
          if (this.items[i].name != 'Sulfuras, Hand of Ragnaros') {
            this.items[i].quality = this.items[i].quality - 1
          }
        }
      } else {
        if (this.items[i].quality < 50) {
          this.items[i].quality = this.items[i].quality + 1
          if (this.items[i].name == 'Backstage passes to a TAFKAL80ETC concert') {
            // Liesbeth TODO: can be written in one if statement
            if (this.items[i].sellIn < 11) {
              if (this.items[i].quality < 50) {
                this.items[i].quality = this.items[i].quality + 1
              }
            }
            if (this.items[i].sellIn < 6) {
              if (this.items[i].quality < 50) {
                this.items[i].quality = this.items[i].quality + 1
              }
            }
          }
        }
      }
      //Liesbeth TODO: should the sellIn date not be updated first? Or together with Q as the normal start.
      if (this.items[i].name != 'Sulfuras, Hand of Ragnaros') {
        this.items[i].sellIn = this.items[i].sellIn - 1;
      }
      if (this.items[i].sellIn < 0) {
        if (this.items[i].name != 'Aged Brie') {
          if (this.items[i].name != 'Backstage passes to a TAFKAL80ETC concert') {
            if (this.items[i].quality > 0) {
              if (this.items[i].name != 'Sulfuras, Hand of Ragnaros') {
                this.items[i].quality = this.items[i].quality - 1
              }
            }
          } else {
            this.items[i].quality = this.items[i].quality - this.items[i].quality
          }
        } else {
          if (this.items[i].quality < 50) {
            this.items[i].quality = this.items[i].quality + 1
          }
        }
      }
    } */
    //TODO: minder geneste if statements
    //TODO: add conjured item SI -1 en Q -2

    return this.items;
  }
}
