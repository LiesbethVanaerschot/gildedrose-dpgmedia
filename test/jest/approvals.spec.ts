import { Item, GildedRose } from '@/gilded-rose';

/**
 * This unit test uses [Jest Snapshot](https://goo.gl/fbAQLP).
 * 
 * There are two test cases here with different styles:
 * <li>"foo" is more similar to the unit test from the 'Java' version
 * <li>"thirtyDays" is more similar to the TextTest from the 'Java' version
 *
 * I suggest choosing one style to develop and deleting the other.
 */

describe('Gilded Rose Approval', () => {

  let gameConsoleOutput: string;
  let originalConsoleLog: (message: any) => void;
  let originalProcessArgv: string[]



  function gameConsoleLog(msg: string) {
    if (msg) {
      gameConsoleOutput += msg;
    }
    gameConsoleOutput += "\n";
  }

  beforeEach(() => {
    // prepare capturing console.log to our own gameConsoleLog.
    gameConsoleOutput = "";
    originalConsoleLog = console.log;
    console.log = gameConsoleLog;
    originalProcessArgv = process.argv;
  });

  afterEach(() => {
    // reset original console.log
    console.log = originalConsoleLog;
    process.argv = originalProcessArgv;
  });


  const items = [
    new Item("+5 Dexterity Vest", 10, 20),
    new Item("Aged Brie", 2, 0),
    new Item("Elixir of the Mongoose", 5, 7),
    new Item("Sulfuras, Hand of Ragnaros", 0, 80),
    new Item("Sulfuras, Hand of Ragnaros", -1, 80),
    new Item("Backstage passes to a TAFKAL80ETC concert", 15, 20),
    new Item("Backstage passes to a TAFKAL80ETC concert", 10, 39),
    new Item("Backstage passes to a TAFKAL80ETC concert", 5, 49),
  
    // This Conjured item does not work properly yet
    new Item("Conjured Mana Cake", 6, 12),
  ];
  
  const gildedRose = new GildedRose(items);
  const days = [...Array(20).keys()];

  test.each(days)('Approvel: Day %d', (d) => {
    gildedRose.updateQuality();
    expect(items).toMatchSnapshot(`day-${d}`);
  })

});
