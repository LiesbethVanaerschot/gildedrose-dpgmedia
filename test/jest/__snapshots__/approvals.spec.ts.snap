// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gilded Rose Approval Approvel: Day 0: day-0 1`] = `
[
  Item {
    "name": "+5 Dexterity Vest",
    "quality": 19,
    "sellIn": 9,
  },
  Item {
    "name": "Aged Brie",
    "quality": 1,
    "sellIn": 1,
  },
  Item {
    "name": "Elixir of the Mongoose",
    "quality": 6,
    "sellIn": 4,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 80,
    "sellIn": 0,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 80,
    "sellIn": -1,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 21,
    "sellIn": 14,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 41,
    "sellIn": 9,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 50,
    "sellIn": 4,
  },
  Item {
    "name": "Conjured Mana Cake",
    "quality": 10,
    "sellIn": 5,
  },
]
`;

exports[`Gilded Rose Approval Approvel: Day 1: day-1 1`] = `
[
  Item {
    "name": "+5 Dexterity Vest",
    "quality": 18,
    "sellIn": 8,
  },
  Item {
    "name": "Aged Brie",
    "quality": 2,
    "sellIn": 0,
  },
  Item {
    "name": "Elixir of the Mongoose",
    "quality": 5,
    "sellIn": 3,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 80,
    "sellIn": 0,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 80,
    "sellIn": -1,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 22,
    "sellIn": 13,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 43,
    "sellIn": 8,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 50,
    "sellIn": 3,
  },
  Item {
    "name": "Conjured Mana Cake",
    "quality": 8,
    "sellIn": 4,
  },
]
`;

exports[`Gilded Rose Approval Approvel: Day 2: day-2 1`] = `
[
  Item {
    "name": "+5 Dexterity Vest",
    "quality": 17,
    "sellIn": 7,
  },
  Item {
    "name": "Aged Brie",
    "quality": 3,
    "sellIn": -1,
  },
  Item {
    "name": "Elixir of the Mongoose",
    "quality": 4,
    "sellIn": 2,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 80,
    "sellIn": 0,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 80,
    "sellIn": -1,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 23,
    "sellIn": 12,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 45,
    "sellIn": 7,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 50,
    "sellIn": 2,
  },
  Item {
    "name": "Conjured Mana Cake",
    "quality": 6,
    "sellIn": 3,
  },
]
`;

exports[`Gilded Rose Approval Approvel: Day 3: day-3 1`] = `
[
  Item {
    "name": "+5 Dexterity Vest",
    "quality": 16,
    "sellIn": 6,
  },
  Item {
    "name": "Aged Brie",
    "quality": 4,
    "sellIn": -2,
  },
  Item {
    "name": "Elixir of the Mongoose",
    "quality": 3,
    "sellIn": 1,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 80,
    "sellIn": 0,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 80,
    "sellIn": -1,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 24,
    "sellIn": 11,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 47,
    "sellIn": 6,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 50,
    "sellIn": 1,
  },
  Item {
    "name": "Conjured Mana Cake",
    "quality": 4,
    "sellIn": 2,
  },
]
`;

exports[`Gilded Rose Approval Approvel: Day 4: day-4 1`] = `
[
  Item {
    "name": "+5 Dexterity Vest",
    "quality": 15,
    "sellIn": 5,
  },
  Item {
    "name": "Aged Brie",
    "quality": 5,
    "sellIn": -3,
  },
  Item {
    "name": "Elixir of the Mongoose",
    "quality": 1,
    "sellIn": 0,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 80,
    "sellIn": 0,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 80,
    "sellIn": -1,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 26,
    "sellIn": 10,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 50,
    "sellIn": 5,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 50,
    "sellIn": 0,
  },
  Item {
    "name": "Conjured Mana Cake",
    "quality": 2,
    "sellIn": 1,
  },
]
`;

exports[`Gilded Rose Approval Approvel: Day 5: day-5 1`] = `
[
  Item {
    "name": "+5 Dexterity Vest",
    "quality": 14,
    "sellIn": 4,
  },
  Item {
    "name": "Aged Brie",
    "quality": 6,
    "sellIn": -4,
  },
  Item {
    "name": "Elixir of the Mongoose",
    "quality": 0,
    "sellIn": -1,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 80,
    "sellIn": 0,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 80,
    "sellIn": -1,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 28,
    "sellIn": 9,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 50,
    "sellIn": 4,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 0,
    "sellIn": -1,
  },
  Item {
    "name": "Conjured Mana Cake",
    "quality": 0,
    "sellIn": 0,
  },
]
`;

exports[`Gilded Rose Approval Approvel: Day 6: day-6 1`] = `
[
  Item {
    "name": "+5 Dexterity Vest",
    "quality": 13,
    "sellIn": 3,
  },
  Item {
    "name": "Aged Brie",
    "quality": 7,
    "sellIn": -5,
  },
  Item {
    "name": "Elixir of the Mongoose",
    "quality": 0,
    "sellIn": -2,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 80,
    "sellIn": 0,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 80,
    "sellIn": -1,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 30,
    "sellIn": 8,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 50,
    "sellIn": 3,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 0,
    "sellIn": -2,
  },
  Item {
    "name": "Conjured Mana Cake",
    "quality": 0,
    "sellIn": -1,
  },
]
`;

exports[`Gilded Rose Approval Approvel: Day 7: day-7 1`] = `
[
  Item {
    "name": "+5 Dexterity Vest",
    "quality": 12,
    "sellIn": 2,
  },
  Item {
    "name": "Aged Brie",
    "quality": 8,
    "sellIn": -6,
  },
  Item {
    "name": "Elixir of the Mongoose",
    "quality": 0,
    "sellIn": -3,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 80,
    "sellIn": 0,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 80,
    "sellIn": -1,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 32,
    "sellIn": 7,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 50,
    "sellIn": 2,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 0,
    "sellIn": -3,
  },
  Item {
    "name": "Conjured Mana Cake",
    "quality": 0,
    "sellIn": -2,
  },
]
`;

exports[`Gilded Rose Approval Approvel: Day 8: day-8 1`] = `
[
  Item {
    "name": "+5 Dexterity Vest",
    "quality": 11,
    "sellIn": 1,
  },
  Item {
    "name": "Aged Brie",
    "quality": 9,
    "sellIn": -7,
  },
  Item {
    "name": "Elixir of the Mongoose",
    "quality": 0,
    "sellIn": -4,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 80,
    "sellIn": 0,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 80,
    "sellIn": -1,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 34,
    "sellIn": 6,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 50,
    "sellIn": 1,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 0,
    "sellIn": -4,
  },
  Item {
    "name": "Conjured Mana Cake",
    "quality": 0,
    "sellIn": -3,
  },
]
`;

exports[`Gilded Rose Approval Approvel: Day 9: day-9 1`] = `
[
  Item {
    "name": "+5 Dexterity Vest",
    "quality": 9,
    "sellIn": 0,
  },
  Item {
    "name": "Aged Brie",
    "quality": 10,
    "sellIn": -8,
  },
  Item {
    "name": "Elixir of the Mongoose",
    "quality": 0,
    "sellIn": -5,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 80,
    "sellIn": 0,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 80,
    "sellIn": -1,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 37,
    "sellIn": 5,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 50,
    "sellIn": 0,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 0,
    "sellIn": -5,
  },
  Item {
    "name": "Conjured Mana Cake",
    "quality": 0,
    "sellIn": -4,
  },
]
`;

exports[`Gilded Rose Approval Approvel: Day 10: day-10 1`] = `
[
  Item {
    "name": "+5 Dexterity Vest",
    "quality": 7,
    "sellIn": -1,
  },
  Item {
    "name": "Aged Brie",
    "quality": 11,
    "sellIn": -9,
  },
  Item {
    "name": "Elixir of the Mongoose",
    "quality": 0,
    "sellIn": -6,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 80,
    "sellIn": 0,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 80,
    "sellIn": -1,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 40,
    "sellIn": 4,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 0,
    "sellIn": -1,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 0,
    "sellIn": -6,
  },
  Item {
    "name": "Conjured Mana Cake",
    "quality": 0,
    "sellIn": -5,
  },
]
`;

exports[`Gilded Rose Approval Approvel: Day 11: day-11 1`] = `
[
  Item {
    "name": "+5 Dexterity Vest",
    "quality": 5,
    "sellIn": -2,
  },
  Item {
    "name": "Aged Brie",
    "quality": 12,
    "sellIn": -10,
  },
  Item {
    "name": "Elixir of the Mongoose",
    "quality": 0,
    "sellIn": -7,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 80,
    "sellIn": 0,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 80,
    "sellIn": -1,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 43,
    "sellIn": 3,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 0,
    "sellIn": -2,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 0,
    "sellIn": -7,
  },
  Item {
    "name": "Conjured Mana Cake",
    "quality": 0,
    "sellIn": -6,
  },
]
`;

exports[`Gilded Rose Approval Approvel: Day 12: day-12 1`] = `
[
  Item {
    "name": "+5 Dexterity Vest",
    "quality": 3,
    "sellIn": -3,
  },
  Item {
    "name": "Aged Brie",
    "quality": 13,
    "sellIn": -11,
  },
  Item {
    "name": "Elixir of the Mongoose",
    "quality": 0,
    "sellIn": -8,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 80,
    "sellIn": 0,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 80,
    "sellIn": -1,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 46,
    "sellIn": 2,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 0,
    "sellIn": -3,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 0,
    "sellIn": -8,
  },
  Item {
    "name": "Conjured Mana Cake",
    "quality": 0,
    "sellIn": -7,
  },
]
`;

exports[`Gilded Rose Approval Approvel: Day 13: day-13 1`] = `
[
  Item {
    "name": "+5 Dexterity Vest",
    "quality": 1,
    "sellIn": -4,
  },
  Item {
    "name": "Aged Brie",
    "quality": 14,
    "sellIn": -12,
  },
  Item {
    "name": "Elixir of the Mongoose",
    "quality": 0,
    "sellIn": -9,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 80,
    "sellIn": 0,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 80,
    "sellIn": -1,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 49,
    "sellIn": 1,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 0,
    "sellIn": -4,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 0,
    "sellIn": -9,
  },
  Item {
    "name": "Conjured Mana Cake",
    "quality": 0,
    "sellIn": -8,
  },
]
`;

exports[`Gilded Rose Approval Approvel: Day 14: day-14 1`] = `
[
  Item {
    "name": "+5 Dexterity Vest",
    "quality": 0,
    "sellIn": -5,
  },
  Item {
    "name": "Aged Brie",
    "quality": 15,
    "sellIn": -13,
  },
  Item {
    "name": "Elixir of the Mongoose",
    "quality": 0,
    "sellIn": -10,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 80,
    "sellIn": 0,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 80,
    "sellIn": -1,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 50,
    "sellIn": 0,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 0,
    "sellIn": -5,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 0,
    "sellIn": -10,
  },
  Item {
    "name": "Conjured Mana Cake",
    "quality": 0,
    "sellIn": -9,
  },
]
`;

exports[`Gilded Rose Approval Approvel: Day 15: day-15 1`] = `
[
  Item {
    "name": "+5 Dexterity Vest",
    "quality": 0,
    "sellIn": -6,
  },
  Item {
    "name": "Aged Brie",
    "quality": 16,
    "sellIn": -14,
  },
  Item {
    "name": "Elixir of the Mongoose",
    "quality": 0,
    "sellIn": -11,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 80,
    "sellIn": 0,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 80,
    "sellIn": -1,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 0,
    "sellIn": -1,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 0,
    "sellIn": -6,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 0,
    "sellIn": -11,
  },
  Item {
    "name": "Conjured Mana Cake",
    "quality": 0,
    "sellIn": -10,
  },
]
`;

exports[`Gilded Rose Approval Approvel: Day 16: day-16 1`] = `
[
  Item {
    "name": "+5 Dexterity Vest",
    "quality": 0,
    "sellIn": -7,
  },
  Item {
    "name": "Aged Brie",
    "quality": 17,
    "sellIn": -15,
  },
  Item {
    "name": "Elixir of the Mongoose",
    "quality": 0,
    "sellIn": -12,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 80,
    "sellIn": 0,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 80,
    "sellIn": -1,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 0,
    "sellIn": -2,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 0,
    "sellIn": -7,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 0,
    "sellIn": -12,
  },
  Item {
    "name": "Conjured Mana Cake",
    "quality": 0,
    "sellIn": -11,
  },
]
`;

exports[`Gilded Rose Approval Approvel: Day 17: day-17 1`] = `
[
  Item {
    "name": "+5 Dexterity Vest",
    "quality": 0,
    "sellIn": -8,
  },
  Item {
    "name": "Aged Brie",
    "quality": 18,
    "sellIn": -16,
  },
  Item {
    "name": "Elixir of the Mongoose",
    "quality": 0,
    "sellIn": -13,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 80,
    "sellIn": 0,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 80,
    "sellIn": -1,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 0,
    "sellIn": -3,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 0,
    "sellIn": -8,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 0,
    "sellIn": -13,
  },
  Item {
    "name": "Conjured Mana Cake",
    "quality": 0,
    "sellIn": -12,
  },
]
`;

exports[`Gilded Rose Approval Approvel: Day 18: day-18 1`] = `
[
  Item {
    "name": "+5 Dexterity Vest",
    "quality": 0,
    "sellIn": -9,
  },
  Item {
    "name": "Aged Brie",
    "quality": 19,
    "sellIn": -17,
  },
  Item {
    "name": "Elixir of the Mongoose",
    "quality": 0,
    "sellIn": -14,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 80,
    "sellIn": 0,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 80,
    "sellIn": -1,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 0,
    "sellIn": -4,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 0,
    "sellIn": -9,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 0,
    "sellIn": -14,
  },
  Item {
    "name": "Conjured Mana Cake",
    "quality": 0,
    "sellIn": -13,
  },
]
`;

exports[`Gilded Rose Approval Approvel: Day 19: day-19 1`] = `
[
  Item {
    "name": "+5 Dexterity Vest",
    "quality": 0,
    "sellIn": -10,
  },
  Item {
    "name": "Aged Brie",
    "quality": 20,
    "sellIn": -18,
  },
  Item {
    "name": "Elixir of the Mongoose",
    "quality": 0,
    "sellIn": -15,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 80,
    "sellIn": 0,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 80,
    "sellIn": -1,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 0,
    "sellIn": -5,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 0,
    "sellIn": -10,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 0,
    "sellIn": -15,
  },
  Item {
    "name": "Conjured Mana Cake",
    "quality": 0,
    "sellIn": -14,
  },
]
`;
